dbms/src/Flash/Executor/DataStreamExecutor.cpp

```diff
@@ -65,11 +69,23 @@ String DataStreamExecutor::toString() const
 
 int DataStreamExecutor::estimateNewThreadCount()
 {
-    return data_stream->estimateNewThreadCount();
+    return estimate_thread_cnt;
 }
 
 RU DataStreamExecutor::collectRequestUnit()
 {
-    return toRU(data_stream->estimateCPUTimeNs());
+    auto origin_cpu_time_ns = data_stream->estimateCPUTimeNs();
```

Comment:
origin_cpu_time_ns = estimate_threads * perThreadCpuTime ?

Reply:
The cputime returned by `BlockInputSrream` is a count of the execution time of each thread. So here we use `cputime / thread_cnt` to get the average cputime of each thread.
However, `per_thread_cputime` is still imprecise and is affected by thread scheduling and `condition_cv.wait`.
The following attempts to eliminate the effects of thread scheduling.